internal/certs/certs.go

```diff
@@ -0,0 +1,153 @@
+package certs
+
+import (
+	"crypto/x509/pkix"
+	"fmt"
+	"github.com/openshift/library-go/pkg/crypto"
+	"github.com/pkg/errors"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"path/filepath"
+	"time"
+)
+
+type CertManager struct {
+	CertificatesDir string
+	userCert        []byte
+	userKey         []byte
+	caBundle        []byte
+}
+type CertInfo struct {
+	keyPath         string
+	certificatePath string
+	commonName      string
+}
+
+func (r *CertManager) GenerateAllCertificates() error {
+	err := r.generateSignerCerts()
```

Comment:
I don't like the name signer certs, let's stick to the OCP convention of being verbose with certificate names as it's very easy to get lost in them - these are kube apiserver serving signing certs